BasementRenovator.py

```diff
@@ -1920,6 +1939,112 @@ def hideWeightPopup(self):
+        self.property2 = int(self.binarySubtype[4:8], 2)
+        self.property3 = int(self.binarySubtype[8:12], 2)
+
+        self.setupList()
+
+        self.layout.addWidget(self.list)
+        self.setLayout(self.layout)
+
+    def setupList(self):
+        self.list = QListWidget()
+        self.list.setViewMode(self.list.ListMode)
+        self.list.setSelectionMode(self.list.ExtendedSelection)
+        self.list.setResizeMode(self.list.Adjust)
+        self.list.setContextMenuPolicy(Qt.CustomContextMenu)
+
+        cursor = QCursor()
+        self.customContextMenu(cursor.pos())
+
+    # @pyqtSlot(int)
+    def changeProperty1(self, var):
+        self.entity.Subtype = 256 * var + 16 * self.property2 + self.property3
+        self.property1 = var
+        mainWindow.dirt()
+        mainWindow.scene.update()
+
+    # @pyqtSlot(int)
+    def changeProperty2(self, var):
+        self.entity.Subtype = 256 * self.property1 + 16 * var + self.property3
+        self.property2 = var
+        mainWindow.dirt()
+        mainWindow.scene.update()
+
+    # @pyqtSlot(int)
+    def changeProperty3(self, var):
+        self.entity.Subtype = 256 * self.property1 + 16 * self.property2 + var
+        self.property3 = var
+        mainWindow.dirt()
+        mainWindow.scene.update()
+
+    # @pyqtSlot(QPoint)
+    def customContextMenu(self, pos):
+        menu = QMenu(self.list)
+
+        if self.entity.Type == 893:
+            # Rotation
+            Rotation = QWidgetAction(menu)
+            r = QComboBox()
+            r.addItem("Clockwise")
+            r.addItem("Counter clockwise")
+            r.setCurrentIndex(self.property1)
+
+            Rotation.setDefaultWidget(r)
+            r.currentIndexChanged.connect(self.changeProperty1)
+            menu.addAction(Rotation)
+        elif self.entity.Type == 3001:
+            # Delay
+            Delay = QWidgetAction(menu)
+            d = QSpinBox()
+            d.setRange(0, 15)
+            d.setPrefix("Delay - ")
+            d.setValue(self.property1)
+
+            Delay.setDefaultWidget(d)
+            d.valueChanged.connect(self.changeProperty1)
+            menu.addAction(Delay)
+
+        # Speed
+        Speed = QWidgetAction(menu)
+        v = QSpinBox()
+        v.setRange(0, 15)
+        v.setPrefix("Speed - ")
+        v.setValue(self.property2)
+
+        Speed.setDefaultWidget(v)
+        v.valueChanged.connect(self.changeProperty2)
+        menu.addAction(Speed)
+
+        # Distance/Angle
+        Distance = QWidgetAction(menu)
+        d = QSpinBox()
+        d.setRange(0, 15)
+        if self.entity.Type == 893:
+            d.setPrefix("Distance - ")
+        elif self.entity.Type == 3001:
+            d.setPrefix("Angle - ")
+        d.setValue(self.property3)
+
+        Distance.setDefaultWidget(d)
+        d.valueChanged.connect(self.changeProperty3)
+        menu.addAction(Distance)
+
+        # End it
+        menu.exec(self.list.mapToGlobal(pos))
+
```

Comment:
regarding the issue with the context menu reappearing everywhere, why not make this whole class extend QMenu and exec it from the mouse handler? that might make it work